File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/ImageAnalyzer.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiImageAnalyzer

class ImageAnalyzer(registrar: CameraXRegistrar) : PigeonApiImageAnalyzer(registrar) {
    override fun pigeon_defaultConstructor(): androidx.camera.core.ImageAnalysis.Analyzer {
        return Impl(this)
    }

    class Impl(private val api: ImageAnalyzer) : androidx.camera.core.ImageAnalysis.Analyzer {
        override fun analyze(image: androidx.camera.core.ImageProxy) {
            api.analyze(this, image) {}
        }
    }
}

File: camerax_ios/ios/Classes/core/ImageAnalyzer.swift
//
//  ImageAnalyzer.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/7/24.
//

import Foundation

class ImageAnalyzer: ImageAnalysis.Analyzer {
    private let api: PigeonApiImageAnalyzerApi
    
    init(api: PigeonApiImageAnalyzerApi) {
        self.api = api
    }
    
    public override func analyze(_ image: ImageProxy) {
        api.analyze(pigeonInstance: self, image: image) { _ in }
    }
}
